### PR TITLE
[address-resolver] remove invalid caches earlier when applicable

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

File: src/core/thread/router_table.cpp [hunk 1]
```diff
@@ -564,6 +564,7 @@ void RouterTable::ProcessTlv(const ThreadRouterMaskTlv &aTlv)
         else
         {
             mAllocatedRouterIds.Remove(routerId);
+            Get<AddressResolver>().Remove(routerId);
```
Comment (simonlingoogle) [inline src/core/thread/router_table.cpp]: Can we even remove the neighbor earlier?

Comment (librasungirl) [reply src/core/thread/router_table.cpp]: Existing implementation has already updated router table accordingly, see https://github.com/openthread/openthread/blob/master/src/core/thread/router_table.cpp#L539

Comment (simonlingoogle) [reply src/core/thread/router_table.cpp]: Why is the logic here not exactly same as in function `void RouterTable::ProcessTlv(const Mle::RouteTlv &aTlv)` ?

File: src/core/thread/router_table.cpp [hunk 2]
```diff
@@ -531,6 +531,7 @@ void RouterTable::ProcessTlv(const Mle::RouteTlv &aTlv)
             RemoveNeighbor(*router);
 
             mAllocatedRouterIds.Remove(routerId);
+            Get<AddressResolver>().Remove(routerId);
```
Comment (simonlingoogle) [inline src/core/thread/router_table.cpp]: `Get<AddressResolver>().Remove(routerId);` is already called by `RemoveNeighbor(*router);` ?

Comment (simonlingoogle) [reply src/core/thread/router_table.cpp]: The current code of handling `RouterTLV` and `ThreadRouterMaskTlv` are not exactly same. Should we just use one same code for processing both TLVs ?

Comment (librasungirl) [reply src/core/thread/router_table.cpp]: Agree that it is not necessary here and there are redundancies in router id set relative process. will close this one submit a separate one for enhancements.

**Pull request closed without merging**